src/docfx/build/metadata/ContributionInfo.cs

```diff
@@ -57,50 +56,56 @@ public ContributionInfo(Docset docset, GitHubUserCache gitHubUserCache)
                     if (!emails.Add(commit.AuthorEmail))
                         continue;
 
-                    if (!resolveGitHubUsers && authorNames.Add(commit.AuthorName))
+                    var contributor = await GetContributor(commit);
+                    if (!excludes.Contains(contributor.Name) && userIds.Add(contributor.Id))
                     {
-                        contributors.Add(new Contributor { DisplayName = commit.AuthorName });
-                        continue;
+                        contributors.Add(contributor);
                     }
-
-                    var (error, user) = await _gitHubUserCache.GetByCommit(commit.AuthorEmail, gitHubOwner, gitHubRepoName, commit.Sha);
-                    if (error != null)
-                        errors.Add(error);
-
-                    if (user != null && !excludes.Contains(user.Login) && logins.Add(user.Login))
-                        contributors.Add(user.ToContributor());
                 }
             }
 
-            if (!string.IsNullOrEmpty(authorName))
+            var author = await GetAuthor();
+            if (author != null)
             {
-                if (resolveGitHubUsers)
-                {
-                    // Remove author from contributors if author name is specified
-                    var (error, author) = await _gitHubUserCache.GetByLogin(authorName);
-                    if (error != null)
-                        errors.Add(error);
+                contributors.RemoveAll(c => c.Id == author.Id);
+            }
 
-                    if (excludes.Contains(authorName))
-                        author = null;
-                    if (author != null)
-                        contributors.RemoveAll(c => c.Id == author.Id.ToString());
+            return (errors, author, contributors, updatedDateTime);
 
-                    return (errors, author?.ToContributor(), contributors, updatedDateTime);
+            async Task<Contributor> GetContributor(GitCommit commit)
+            {
+                if (!resolveGitHubUsers)
+                {
+                    return new Contributor { DisplayName = commit.AuthorName, Id = commit.AuthorEmail };
```

Comment:
> Id = commit.AuthorEmail [](start = 77, length = 24)

Will template consume Id? but actually it's a email.

Reply:
no, it won't